src/logging.rs

```diff
@@ -17,32 +17,40 @@ thread_local! {
 pub mod trace {
     extern crate tracing_log;
     use super::{clock, Context as LegacyContext, CONTEXT as LEGACY_CONTEXT};
-    use std::{env, error, fmt, time::Instant};
+
+    use std::{env, error, fmt, str, time::Instant};
     pub use tracing::*;
     pub use tracing_fmt::*;
 
     type SubscriberBuilder = Builder<default::NewRecorder, Format, filter::EnvFilter>;
     pub type Error = Box<error::Error + Send + Sync + 'static>;
 
+    #[derive(Clone)]
+    pub struct LevelHandle {
```

Comment:
Could we call this `TraceLevel` so it doesn't need to be imported `as` in `src/app/admin/mod.rs`?

Reply:
Meant to actually delete this comment on my review. After looking more at how `LevelHandle` is used here, I see the naming benefit of referring to it as a handle in this file. In other modules where it's easier to think of it just as the level makes sense.

Reply:
I'm happy to just call this `TraceLevel` here, that might be clearer?

Reply:
I think keeping it `LevelHandle` is okay! `TraceLevel` is helpful in other files, but it's a handle everywhere in this file, so I'm happy with the naming.